src/ulid/exceptions.cr

```diff
@@ -0,0 +1,3 @@
+class IncorrectLength < Exception; end
+
+class InvalidChars < Exception; end
```

Comment:
Having these as seperate error types provides little useful information. For a user, the only concern is it's either a valid ULID, or not. The reason for an invalidity can then be contained in the associated exception message.

These are also currently sitting in the top-level namespace. Definitely a good idea to nest any exception(s) that are defined within the `ULID` module.

Reply:
Definitely agree with this one

Reply:
So instead of: 
`raise IncorrectLength.new("ULIDs must be 26 characters.") unless ulid.size == TIME_LEN + RANDOM_LEN`
should I
`raise Exception.new("ULIDs must be 26 characters.") unless ulid.size == TIME_LEN + RANDOM_LEN` ?

Or keep custom exceptions just inside ULID module?

Reply:
I think custom exception inside of the ULID module makes the most sense to me

Reply:
You could also move the message inside of the error definition

Reply:
okay, fixed this up

I'm a little unsure about the file structure
I've just put the exception classes in the ULID module under the `.validate!` method. Is there a more appropriate place?

```crystal
 def validate!(ulid : String) : Nil
    # Incorrect length
    raise IncorrectLength.new() unless ulid.size == TIME_LEN + RANDOM_LEN
    # Invalid chars
    raise InvalidChars.new() unless ulid.upcase.chars.all? &.in?(ENCODING)

    nil
  end

  class IncorrectLength < Exception
    def initialize(message = "ULIDs must be 26 characters.")
      super(message)
    end
  end

  class InvalidChars < Exception
    def initialize(message = "Invalid characters found.")
      super(message)
    end
  end
  ```

Reply:
With the design of this you want to consider the how someone will use this lib. Calls to `.validate!` may occur somewhere deep within a call stack, with exception handling for this possible a little further away.

With the current approach a user must either create a [pokemon exception handler](http://wiki.c2.com/?PokemonExceptionHandling), which could capture any number of horrible things computers do, or individually catch both `ULID::IncorrectLength` _and_ `ULID::InvalidChars`. In reality all they want to know about is `ULID::Invalid`.

iIf you are modelling something with more diverse failure options a good pattern is a base exception for the domain that you are working in (`Foo::Error` or similar), and then more explicit error types that inherit from this. This was the base error can be caught for a general handler, or more specific ones where suitable too. In this case though I'd just have a single error type to keep things simple.